File: javascript/calculator.js
class SafeCalculator {
  constructor (authorizer) {
    this.authorizer = authorizer
  }

  add (a, b) {
    const authorized = this.authorizer.authorize()
    // Bug! Should be `if(!authorized)`
    if (authorized) {
      throw new Error('Not authorized')
    }
    return a / b
  }
}

module.exports = SafeCalculator

File: c/src/calculator.c
#include "calculator.h"
#include "authorizer.h"

bool calculator_add(int left, int right, int *result)
{
    // BUG! Should be calculation_is_authorized() without the !
    if (!calculation_is_authorized()) {
        *result = left + right;
        return true;
    } else {
        return false;
    }
}
